poker.js

```diff
@@ -0,0 +1,442 @@
+  let threeOfAKindCard =
+    Object.keys(pairValue2)[Object.values(pairValue2).indexOf(3)];
+  let pairCard = Object.keys(pairValue2)[Object.values(pairValue2).indexOf(2)];
+  let duplicates = Object.keys(pairValue2).filter(
+    (value) => pairValue2[value] === 2
+  );
+
+  if (isARoyalFlush(hand.player2Cards)) {
+    player2Score = combination.royalFlush;
+    player2ScoreCard.push(combination.royalFlush);
+  } else if (isStraightFlush(hand.player2Cards)) {
+    player2Score = combination.straightFlush + isAHighCard(getCardValue);
+    player2ScoreCard.push(player2Score);
+  } else if (isFourOfAKind(hand.player2Cards)) {
+    player2Score = combination.fourOfAKind + parseInt(fourOfAKindCard);
+    player2ScoreCard.push(player2Score);
+  } else if (isFullHouse(hand.player2Cards)) {
+    player2Score = combination.fullHouse + parseInt(threeOfAKindCard);
+    player2ScoreCard.push(player2Score);
+  } else if (isAFlush(hand.player2Cards)) {
+    player2Score = combination.flush + isAHighCard(getCardValue);
+    player2ScoreCard.push(player2Score);
+  } else if (isAStraight(hand.player2Cards)) {
+    player2Score = combination.straight + isAHighCard(getCardValue);
+    player2ScoreCard.push(player2Score);
+  } else if (isThreeOfAKind(hand.player2Cards)) {
+    player2Score = combination.threeOfAKind + parseInt(threeOfAKindCard);
+    player2ScoreCard.push(player2Score);
+  } else if (isTwoPairs(hand.player2Cards)) {
+    player2Score =
+      combination.twoPairs + parseInt(sumOfDuplicatedItems(duplicates));
+    player2ScoreCard.push(player2Score);
+  } else if (isAPair(hand.player2Cards)) {
+    player2Score = combination.pair + parseInt(pairCard);
+    player2ScoreCard.push(player2Score);
+  } else {
+    player2Score = isAHighCard(getCardValue);
+    player2ScoreCard.push(player2Score);
+  }
+  return player2Score;
+};
+
+const isARoyalFlush = (cards) => {
+  /**
+   * Ten, Jack, Queen, King and Ace in the same suit
+   */
+
+  //check the cards on hand is same as royalCards
+  let royalCards = ["T", "J", "Q", "K", "A"];
+  let getCards = getCardValues(cards);
+  let checkCardValues = royalCards.every((item) => getCards.includes(item));
+
+  //if suitsCount = 5 and cards on Hand = royalCards, then it is a royal flush
+  if (isAFlush(cards) && checkCardValues === true) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const isStraightFlush = (cards) => {
+  /**
+   * All five cards in consecutive value order, with the same suit
+   */
+
+  if (isAFlush(cards) && checkCardsIfConsecutiveOrder(cards) === true) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const isFourOfAKind = (cards) => {
+  /**
+   * Four cards of same value
+   */
+
+  //counting the number of occurences for each card value
+  const fourCount = cardCount(cards);
+
+  //check if one of the card values count up to 4
+  if (Object.values(fourCount).includes(4)) {
+    return true;
+  } else {
+    return false;
+  }
+};
```

Comment:
if you have a structure like
```js
if (condition) {
    return true;
} else {
    return false;
}
```

you can replace the entire if statement with:
```js
return condition;
```

because the condition already returns true or false

Reply:
```js
const isFourOfAKind = (cards) => {
  /**
   * Four cards of same value
   */

  //counting the number of occurences for each card value
  const fourCount = cardCount(cards);

  //check if one of the card values count up to 4
  return Object.values(fourCount).includes(4);
};
```

You can do the same for most of the other functions in this file

Reply:
done